tests/api/v2/handlers/test_agents_api.py

```diff
@@ -0,0 +1,222 @@
+    return {
+        'app.contact.dns.domain': 'mycaldera.caldera',
+        'app.contact.dns.socket': '0.0.0.0:8853',
+        'app.contact.ftp.host': '0.0.0.0',
+        'app.contact.http': 'http://0.0.0.0:8888',
+        'app.contact.tcp': '0.0.0.0:7010'
+    }
+
+
+@pytest.fixture
+def agent_config():
+    return {
+        'deadman_abilities': [],
+        'implant_name': 'test',
+        'sleep_max': 6,
+        'sleep_min': 3,
+        'watchdog': 0
+    }
+
+
+@pytest.fixture
+def combined_config(agent_config, app_config):
+    app_config.update({f'agents.{k}': v for k, v in agent_config.items()})
+    return app_config
+
+
+class TestAgentsApi:
+    async def test_get_agents(self, api_v2_client, api_cookies, test_agent):
+        resp = await api_v2_client.get('/api/v2/agents', cookies=api_cookies)
+        assert resp.status == HTTPStatus.OK
+        agents_list = await resp.json()
+        assert len(agents_list) == 1
+        agents_dict = agents_list[0]
+        assert agents_dict == test_agent.schema.dump(test_agent)
+
+    async def test_unauthorized_get_agents(self, api_v2_client, test_agent):
+        resp = await api_v2_client.get('/api/v2/agents')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_get_agent_by_id(self, api_v2_client, api_cookies, test_agent):
+        resp = await api_v2_client.get('/api/v2/agents/123', cookies=api_cookies)
+        assert resp.status == HTTPStatus.OK
+        agent_dict = await resp.json()
+        assert agent_dict == test_agent.schema.dump(test_agent)
+
+    async def test_unauthorized_get_agent_by_id(self, api_v2_client, test_agent):
+        resp = await api_v2_client.get('/api/v2/agents/123')
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_nonexistent_get_agent_by_id(self, api_v2_client, api_cookies):
+        resp = await api_v2_client.get('/api/v2/agents/999', cookies=api_cookies)
+        assert resp.status == HTTPStatus.NOT_FOUND
+
+    async def test_create_agent(self, api_v2_client, api_cookies, new_agent_payload):
+        resp = await api_v2_client.post('/api/v2/agents', cookies=api_cookies, json=new_agent_payload)
+        assert resp.status == HTTPStatus.OK
+        agent_dict = await resp.json()
+        assert agent_dict['sleep_min'] == new_agent_payload['sleep_min']
+        assert agent_dict['sleep_max'] == new_agent_payload['sleep_max']
+        assert agent_dict['paw'] == new_agent_payload['paw']
+        assert agent_dict['executors'] == new_agent_payload['executors']
+
+    async def test_unauthorized_create_agent(self, api_v2_client, new_agent_payload):
+        resp = await api_v2_client.post('/api/v2/agents', json=new_agent_payload)
+        assert resp.status == HTTPStatus.UNAUTHORIZED
+
+    async def test_update_agent(self, api_v2_client, api_cookies, test_agent, updated_agent_payload):
+        resp = await api_v2_client.patch('/api/v2/agents/123', cookies=api_cookies, json=updated_agent_payload)
```

Comment:
I'm wondering if it would be more appropriate to send only the fields that are being updated. Otherwise this is kind of like mimicking `put`